test/distributed/org/apache/cassandra/distributed/test/auth/MutualTlsCertificateValidityPeriodTest.java

```diff
@@ -0,0 +1,329 @@
+                                 .set("client_encryption_options.keystore", serverKeystorePath.toString())
+                                 .set("client_encryption_options.keystore_password", SERVER_KEYSTORE_PASSWORD)
+                                 .set("client_encryption_options.truststore", truststorePath.toString())
+                                 .set("client_encryption_options.truststore_password", SERVER_TRUSTSTORE_PASSWORD)
+                                 .set("client_encryption_options.require_endpoint_verification", "false")
+                                 .set("client_encryption_options.max_certificate_validity_period", "30d")
+                                 .set("client_encryption_options.certificate_validity_warn_threshold", "5d")
+                                 .with(Feature.NATIVE_PROTOCOL, Feature.GOSSIP));
+        CLUSTER = builder.start();
+
+        configureIdentity();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception
+    {
+        LOGGER.info("[test step : @AfterClass] teardown");
+        if (CLUSTER != null)
+            CLUSTER.close();
+    }
+
+    @Test
+    public void testExpiringCertificate() throws Exception
+    {
+        LOGGER.info("[test step : @Test] testExpiringCertificate");
+        Path clientKeystorePath = generateClientCertificate(null);
+
+        com.datastax.driver.core.Cluster driver = JavaDriverUtils.create(CLUSTER, null, b -> b.withSSL(getSSLOptions(clientKeystorePath)));
+
+        testWithDriver(driver, (Session session) -> {
+            ResultSet clientView = session.execute(new SimpleStatement("SELECT * FROM system_views.clients"));
+            Assertions.assertThat(clientView).isNotNull().isNotEmpty();
+
+            Optional<Row> thisClient = StreamSupport.stream(clientView.spliterator(), false)
+                                                    .filter(row -> "cassandra_ssl_test".equals(row.getString("username")))
+                                                    .findFirst();
+
+            Assertions.assertThat(thisClient).isPresent();
+            Row row = thisClient.get();
+            Map<String, String> authenticationMetadata = row.getMap("authentication_metadata", String.class, String.class);
+
+            Assertions.assertThat(authenticationMetadata).isNotNull().hasSize(1)
+                      .containsKey("identity")
+                      .extractingByKey("identity", as(InstanceOfAssertFactories.STRING)).isEqualTo(IDENTITY);
+            Assertions.assertThat(row.getString("authentication_mode")).isEqualTo("MutualTls");
+        });
+    }
+
+    @Test
+    public void testCertificateReachingMaxValidityPeriod() throws Exception
+    {
+        LOGGER.info("[test step : @Test] testCertificateReachingMaxValidityPeriod");
+        Path clientKeystorePath = generateClientCertificate(b -> b.notBefore(Instant.now().minus(3, ChronoUnit.DAYS))
```

Comment:
`client_encryption_options.max_certificate_validity_period` is set to 30 days, shouldn't this be a larger range?